vllm/transformers_utils/runai_utils.py

```diff
@@ -63,9 +63,6 @@ def __init__(self) -> None:
 
         self.dir = tempfile.mkdtemp()
 
-    def __del__(self):
```

Comment:
I think we should refactor the usage of ObjectStorageModel to cache downloaded files, based on the model name. This will allow:
1) The same model directory to be re-used upon vLLM restart / re-invocation. This will save latency if metadata files have already been downloaded
2) Re-use of the torch cache

Then we don't need to worry about cleaning up the directory, as data can be downloaded to a cache location (eg: in `$VLLM_ASSETS_CACHE/model_streamer/$MODEL_NAME`